### PR TITLE
fix: build on windows

### LINKED ISSUE #786: Failed to install on Windows
```
error C:\Users\Jack\Workspace\endo\node_modules\@endo\endo: Command failed.
Exit code: 1
Command: node src/postinstall.js
Arguments:
Directory: C:\Users\Jack\Workspace\endo\node_modules\@endo\endo
Output:
node:fs:582
  handleErrorFromBinding(ctx);
  ^

Error: ENOENT: no such file or directory, open '/C:/Users/Jack/Workspace/endo/packages/endo/mitm/node'
```

This script `@endo/endo/src/postinstall.js` is *nix only.

### PR DESCRIPTION
**Pull request by Jack-Works**
close #786

### CODE REVIEW

**Comment by Jack-Works** [comment]
hi @mhofman can you review this?

**Comment by kumavis** [comment]
TIL `fileURLToPath`
chqnges look reasonable but im not very familiar with windows

**Comment by Jack-Works** [comment]
> TIL `fileURLToPath`
> changes look reasonable but im not very familiar with windows

Previously it is using a relative URL as a path, which is almost the same thing on *nix system. I have to convert them to the path so they can work on Windows too.

**Comment by mhofman** [comment]
@Jack-Works, I actually don't have a proper Win32 dev environment, only WSL. Is there any chance you could update CI tests to run on Windows to make sure we don't break any of your changes by mistake?

File: packages/endo/src/postinstall.js [hunk 1]
```diff
@@ -6,14 +6,35 @@
...
 set -ueo pipefail
 ${node} -r ${lockdown} "$@"`;
 
-fs.writeFileSync(mitm, script, 'utf-8');
+const win32Script = `
+@IF EXIST "%~dp0\\node.exe" (
+    "%~dp0\\node.exe" -r ${lockdown} -r %*
+) ELSE (
+    @SETLOCAL
+    @SET PATHEXT=%PATHEXT:;.JS;=;%
```
Comment (mhofman) [inline packages/endo/src/postinstall.js:30]: Why is setting `PATHEXT` necessary?

File: packages/ses/scripts/bundle.js [hunk 2]
```diff
@@ -2,20 +2,21 @@ import '../index.js';
 import fs from 'fs';
 import { makeBundle } from '@endo/compartment-mapper/bundle.js';
 import terser from 'terser';
+import { fileURLToPath, pathToFileURL } from 'url';
 
-const resolve = (rel, abs) => new URL(rel, abs).toString();
-const root = resolve('..', import.meta.url);
+const resolve = (rel, abs) => fileURLToPath(new URL(rel, abs));
```
Comment (mhofman) [inline packages/ses/scripts/bundle.js:7]: It feels like `resolve` is not really longer necessary and can be inlined where used?

Comment (kriskowal) [reply packages/ses/scripts/bundle.js:7]: I could go either way on this. It is a mouthful, but I’ve inlined it elsewhere habitually.

File: packages/endo/src/postinstall.js (hunk 1 above)
Comment (Jack-Works) [reply packages/endo/src/postinstall.js:30]: Have no idea, `cmd` is a horrifying language. I'm copying this from how `npm` generates cmd for bin files.

**Comment by Jack-Works** [comment]
hi! I found a problem with Windows test.

`packages/compartment-mapper/test/fixtures-0/node_modules/app/evan` is a symbolic link which cannot be handled correctly on Git for Windows.

image

It has been checkout as a normal file on Windows by default.

I tried to recreate the symbolic link manually by

- Symbolic link: `mklink /D packages\compartment-mapper\test\fixtures-0\node_modules\app\node_modules\evan\ C:\Users\Jack\Workspace\endo\packages\compartment-mapper\test\fixtures-1\node_modules\evan\`
- Or Directory junction `mklink /J packages\compartment-mapper\test\fixtures-0\node_modules\app\node_modules\evan packages\compartment-mapper\test\fixtures-1\node_modules\evan`

Test failed with:

```
message: 'Cannot find dependency evankin for file:///C:/Users/Jack/Workspace/endo/packages/compartment-mapper/test/fixtures-0/node_modules/app/node_modules/evan/'
```

It seems like it failed to resolve dependency of `eval` (resolve back to real position and find it's real node_modules).

Any suggestions?

**Comment by kriskowal** [comment]
Just FYI, this is on my plate and I will have to look into it soon.

**Comment by kriskowal** [comment]
I’ve subsumed @Jack-Works’s commits from this into #948 and pushed thru to working end-to-end.

**Pull request closed without merging**